pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -83,45 +299,49 @@ func (ssc *defaultStatefulSetControl) UpdateStatefulSet(set *apps.StatefulSet, p
 		// If the ordinal could not be parsed (ord < 0), ignore the Pod.
 	}
 
-	// for any empty indices in the sequence [0,set.Spec.Replicas) create a new Pod
+	// for any empty indices in the sequence [0,set.Spec.Replicas) create a new Pod at the correct revision
 	for ord := 0; ord < replicaCount; ord++ {
 		if replicas[ord] == nil {
-			replicas[ord] = newStatefulSetPod(set, ord)
+			replicas[ord] = newVersionedStatefulSetPod(currentSet, updateSet, currentRevision.Name, updateRevision.Name, ord)
 		}
 	}
 
-	// count the number of unhealthy pods
+	// sort the condemned Pods by their ordinals
+	sort.Sort(ascendingOrdinal(condemned))
+
+	// find the first unhealthy Pod
```

Comment:
nit: why do we care about the firstUnhealthyPod, when you get time, may be add a comment in subsequent PR